### ISSUE TITLE
godot 4 hp reverb gen 1 Input Problems 

### ISSUE THREAD

**Issue by pietru2004** [GodotVR/godot-xr-tools#327]
touchpad and menu btn seam to have no binds...

(Issue https://github.com/GodotVR/godot-xr-tools/issues/313 related)

Controler has:
- joystick
- touchpad
- trigger
- grip
- menu button
- windows menu btn (not bindable if I am correct)

**Comment by BastiaanOlij** [comment]
This should be raised on the Godot repo itself, not here. 

That said, looking at the OpenXR spec only the G2 controllers are supported: https://registry.khronos.org/OpenXR/specs/1.0/html/xrspec.html#XR_EXT_hp_mixed_reality_controller

These only have thumbstick defined, so this looks like an issue to raise with OpenXR (which I'll do)

**Comment by BastiaanOlij** [comment]
Hang on, did the G1 just come with standard WMR controllers? Sounds like there is an issue in SteamVR thinking you have G2 controllers?

**Comment by pietru2004** [comment]
Personaly I am recal steam tutorial using/showing all butrons, but in any case here is cotroller pic

![20230112_090159.jpg](https://user-images.githubusercontent.com/51274570/212011332-314d2a52-57ac-49cf-bb5e-5b4165dbf43f.jpg)

**Comment by BastiaanOlij** [comment]
Yeah that is the original MR controller, so my guess is that SteamVR is automatically assuming you have the reverb controllers because you're using a Reverb headset. It should be using the normal MR interaction profile which does have all the proper inputs. 

The HPMR interaction profile purely relates to these controllers:
![image](https://user-images.githubusercontent.com/1945449/212580007-ea2383ba-3e04-4320-b2cd-4ac4e29d7c30.png)

**Comment by BastiaanOlij** [comment]
Little bit more feedback on this. So far it seems that when using WMR over SteamVR, WMR is reporting the use of G2 controllers if you're using a Reverb headset, even if you're using the default MR controllers.
This is an issue either in the WMR drivers or in SteamVR, it's being looked into.

**Comment by pietru2004** [comment]
I recall steam vr tutorial showing correct controller.

**Comment by pietru2004** [comment]
@BastiaanOlij I found out my controlers name
https://www.ifixit.com/Device/Dell_VRC100-R

**Comment by BastiaanOlij** [comment]
Hey @pietru2004 ,

This is a bit of a blast from the past. :)

If I recall correctly the problem is that the reverb came out with newer controllers and for some reason the WMR system is always reporting those controller profiles when a Reverb headset is used, even if you're using the old style WMR controllers that came with nearly all WMR headsets (you're showing the DELL ones but I have the same ones with my Acer). 

Obviously that will make the mapping fail because the reverb controllers have very different controls. Just for comparison: these are the controllers it's expecting to see:
![image](https://github.com/GodotVR/godot-xr-tools/assets/1945449/b527c303-6221-4d59-ab85-0835e76ecc22)

From what I was told this was a bug in either the WMR drivers for Steam or in WMR itself, though as far as I'm aware this is working correctly over OpenVR and not OpenXR, am I correct in that?

I'll see if I can nudge someone to have another look.

**Comment by pietru2004** [comment]
I'm not sure all I recall is that steamVR has correct bindings for my controler - it shows up correctly and works with tutorial...
As far as I can tell I have selected SteamVR OpenXR env. if I rebind controler inputs they seam to work (ax and by button to touchpad zones - 5 zones available, edges and center).
I have those options for bindings
![obraz](https://github.com/GodotVR/godot-xr-tools/assets/51274570/ca9fd4ab-83bd-4122-82d0-8ea28d9c53b9)

defaults are
trigger:
![obraz](https://github.com/GodotVR/godot-xr-tools/assets/51274570/81012681-63df-456a-87b3-e6bbbd3bab09)
touchpad:
![obraz](https://github.com/GodotVR/godot-xr-tools/assets/51274570/c640da35-bcc1-45e0-8b77-62a3911965b1)
grip:
![obraz](https://github.com/GodotVR/godot-xr-tools/assets/51274570/91ebae88-fad7-4fa7-a6b6-1097cf403dd3)
menu and joystick:
![obraz](https://github.com/GodotVR/godot-xr-tools/assets/51274570/e5e40551-45bd-44e3-b232-31ccbc960748)


Kliknięcie means Click
Pozycja means position
Dotknięcie means touch

**Comment by BastiaanOlij** [comment]
Hmmm, weird, if everything is showing correctly in the SteamVR binding menu that means it is applying the right bindings, I don't get why it doesn't work then, unless it doesn't allow you to use the menu button because it's reserved.

**Comment by BastiaanOlij** [comment]
It's possible because you have created custom bindings that SteamVR starts ignoring our bindings? I don't know, it's hard to say without having the same hardware available...

**Comment by pietru2004** [comment]
By default joystick click opens steam menu, the ax and by are not binded anywhere
It could have touchpad to ax  by conversion

**Comment by BastiaanOlij** [comment]
Hey @pietru2004 , are you still having issues with this? So far I've not gotten any useful information about it other than Valve mentioning this is an issue with the data they are getting from WMR (and it expecting the newer G2 controllers, not the old WMR controllers).

I'd like to close this issue as its not really related to XR Tools.

**Comment by pietru2004** [comment]
I seam to be able to override this problem using steam input bindings, but other than that I couldn't find other way. If we hit dead end on ossue I think you can close it.

**Comment by pietru2004** [comment]
@BastiaanOlij this can be closed - gotten different headset and I think it won't be issue no, at least for me.

**Comment by BastiaanOlij** [comment]
Thanks @pietru2004, I think with Microsoft sadly winding back WMR support anyway, this is indeed not something that will likely be resolved. Hopefully HP will follow up with hardware based directly off of SteamVR.
